pins/drivers.py

```diff
@@ -15,14 +15,38 @@
 
 
 def _assert_is_pandas_df(x, file_type: str) -> None:
-    import pandas as pd
+    df_family = _get_df_family(x)
 
-    if not isinstance(x, pd.DataFrame):
+    if df_family != "pandas":
         raise NotImplementedError(
             f"Currently only pandas.DataFrame can be saved as type {file_type!r}."
         )
 
 
+def _get_df_family(df) -> Literal["pandas", "polars"]:
```

Comment:
```suggestion
def _get_df_family(df, file_type: str) -> Literal["pandas", "polars"]:
```

Reply:
Good idea, I think I see a nice abstraction for this...